arch/risc-v/src/mpfs/Kconfig

```diff
@@ -511,34 +511,65 @@ config MPFS_COREPWM1_NCHANNELS
 
 comment "CAN-FD Options"
 
-config MPFS_CANFD
+config MPFS_HAVE_CANFD
 	bool "CAN FD"
 	select ARCH_HAVE_CAN_ERRORS
 	select NET_CAN_HAVE_CANFD
 	select NET_CAN_EXTID
 	select NET_CAN_HAVE_TX_DEADLINE
 	default n
 
-config MPFS_CANFD_BASE
-	hex "Base address for the instance"
+config MPFS_CANFD0
+	bool "MPFS FPGA CANFD0 IP block configured"
+	default n
+	depends on MPFS_HAVE_CANFD
+
+config MPFS_CANFD_BASE0
+	hex "Base address for the CANFD0 instance"
 	default 0x46000000
-	depends on MPFS_CANFD
+	depends on MPFS_CANFD0
+
+config MPFS_CANFD_CLK0
+	int "Clock frequency of the CANFD0 block (Hz)"
+	default 62500000
+	range 1000000 100000000
+	depends on MPFS_CANFD0
+
+config MPFS_CANFD_ARBI_BITRATE0
+	int "CANFD0 Arbitration phase bitrate"
+	default 1000000
+	depends on MPFS_CANFD0
+
+config MPFS_CANFD_DATA_BITRATE0
+	int "CANFD0 Data phase bitrate"
+	default 4000000
+	depends on MPFS_CANFD0
+
+config MPFS_CANFD1
+	bool "MPFS FPGA CANFD1 IP block configured"
+	default n
+	depends on MPFS_HAVE_CANFD
+
+config MPFS_CANFD_BASE1
+	hex "Base address for the CANFD1 instance"
+	default 0x47000000
+	depends on MPFS_CANFD1
 
-config MPFS_CANFD_CLK
-	int "Clock frequency of the CANFD block (Hz)"
+config MPFS_CANFD_CLK1
```

Comment:
I think this is always the same as the FPGA clock, so it doesn't need to be defconfig.. It fits better to board.h, but there is already a define for it, see CorePWM for example.

Also note that it was changed to 50000000 in out fpga designs

Reply:
yes, I rebased and made the change to use peripheral clock config